app/src/main/java/org/mozilla/fenix/settings/quicksettings/QuickSettingsFragmentState.kt

```diff
@@ -64,6 +65,15 @@ data class WebsiteInfoState(
             return WebsiteInfoState(websiteUrl, websiteTitle, uiValues, certificateName)
         }
     }
+
+    @Suppress("UnstableApiUsage")
+    fun baseDomain(): String {
+        val host = websiteUrl.toUri().host.toString()
+        if (!InternetDomainName.isValid(host)) {
+            return host
+        }
+        return InternetDomainName.from(host).topPrivateDomain().toString()
```

Comment:
I think we could replace this call with [getOrigin](https://github.com/mozilla-mobile/android-components/blob/b7d7569909decf292214b4a9c2c1d81c3641b91f/components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/String.kt#L157)

Reply:
We could pass directly the host as the inter API only removes by host see [clearData](https://github.com/mozilla-mobile/android-components/blob/b7d7569909decf292214b4a9c2c1d81c3641b91f/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt#L518)

Reply:
Thanks for your comments!

The key part here is [`topPrivateDomain()`](https://guava.dev/releases/snapshot-jre/api/docs/com/google/common/net/InternetDomainName.html#topPrivateDomain()), that returns the part of the domain that is one level beneath the public suffix, and this is the tricky part. Most public suffixes are single strings as `.com`, `.org` and so on, but then there are country specific codes with secondary levels, as `.co.uk`. Even some services have specific public suffixes such as `.blogspot.com`.

Why do we need the part beneath the public suffix? Looking at the implementation in Firefox for desktop, you can see that in the list of cookies, all of them are domains with a single name plus the public suffix, no subdomains. If you click the clear site data button, you will see that the confirmation dialog always shows a top level domain without the subdomains.

I tried to pass the host directly to `clearData`, and it worked for websites without subdomains, but it didn't work for websites with subdomains. For example it works when browsing `https://elpais.com`, and clearing `elpais.com`, but it doesn't when browsing `https://www.theguardian.com` and clearing `www.theguardian.com`, for the second case to work, `clearData` must be called with `theguardian.com`.

I started trying to implement my own logic to get the top level domain, but I found it error prone, for example for the cases like `.co.uk` or `.blogspot.com`. Then I looked for implementations, and I couldn't find if the one in Gecko is exposed for android (I am guessing that Gecko uses something based on this [EffectiveTLDService](https://searchfox.org/mozilla-central/rev/fb8d77331582639ea6848a61dd8ee812fac31b77/netwerk/dns/nsEffectiveTLDService.cpp)). This is why I finally imported it from Guava.

I see some possible follow ups on this:
* I am missing something and there is a simpler way of finding the proper host for `clearData` :slightly_smiling_face: 
* We find some way of exposing the implementation in Gecko.
* We try to implement a `topPrivateDomain()` method based for example on the [Public Suffix list](https://publicsuffix.org/list/) maintained by the Mozilla Foundation.
* We continue using Guava till we find a better option.

Reply:
Maybe @agi from the GeckoView team could help us to know if there is something similar exposed from Gecko to Android via GeckoView.

Reply:
@jsoriano just to make you aware it may take some time for Agi to response as it's a holiday for him.

Reply:
@jsoriano after chatting with @agi, he mentioned that the GeckoView team will be exposing an API for cleaning sites data by using a [GeckoSession instead of a text based URL](https://bugzilla.mozilla.org/show_bug.cgi?id=1743639) this way we won't need to use the Guava APIs, he also mentioned that in the meantime we could give a try to [runtime.storageController.clearDataFromBaseDomain](https://searchfox.org/mozilla-central/source/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/StorageController.java#135). To test it it will required some changes on AC for you to see it from Fenix, you will need to expose a new function on [AC](https://github.com/mozilla-mobile/android-components/blob/b7d7569909decf292214b4a9c2c1d81c3641b91f/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt#L518)

Reply:
Just as a side note, we do have `lib-publicsuffixlist` in A-C. :)

Reply:
😅 Awesome thanks @pocmo. 
 @jsorian In this case we could use `lib-publicsuffixlist `. We are already using [it in multiple places ](https://github.com/mozilla-mobile/fenix/search?q=publicSuffixList) in Fenix :)

Reply:
Thanks for the hint! Changed in https://github.com/mozilla-mobile/fenix/pull/22455/commits/c6887cb79afe4880159884e0c3d1645a40b958bb. It seems to work as expected.

It would be still interesting if gecko could expose information about the current session, this would allow to show the option to clear data only if there is some data.